integration-cli/docker_utils_test.go

```diff
@@ -719,11 +710,80 @@ func buildImageWithStdoutStderr(name, dockerfile string, useCache bool, buildFla
 	return id, result.Stdout(), result.Stderr(), nil
 }
 
+func buildImageSuccessfully(c *check.C, name string, cmdOperators ...func(*icmd.Cmd) func()) {
+	buildImageNew(name, cmdOperators...).Assert(c, icmd.Success)
+}
+
+// FIXME(vdemeester) rename this buildImage once deprecated buildImage is no more
+func buildImageNew(name string, cmdOperators ...func(*icmd.Cmd) func()) *icmd.Result {
+	cmd := icmd.Command(dockerBinary, "build", "-t", name)
+	for _, op := range cmdOperators {
+		deferFn := op(&cmd)
+		if deferFn != nil {
+			defer deferFn()
+		}
+	}
+	return icmd.RunCmd(cmd)
+}
+
+func withContext(c *check.C, contextOperators ...func(*FakeContext) error) func(*icmd.Cmd) func() {
+	ctx, err := fakeContextFromNewTempDir()
+	if err != nil {
+		c.Fatalf("error creating build context : %v", err)
+	}
+	for _, op := range contextOperators {
+		if err := op(ctx); err != nil {
+			c.Fatal(err)
+		}
+	}
+	return func(cmd *icmd.Cmd) func() {
+		cmd.Dir = ctx.Dir
+		cmd.Command = append(cmd.Command, ".")
+		return closeCtx(c, ctx)
+	}
+}
+
+func withBuildFlags(flags ...string) func(*icmd.Cmd) func() {
+	return func(cmd *icmd.Cmd) func() {
+		cmd.Command = append(cmd.Command, flags...)
+		return nil
+	}
+}
+
+func withoutCache(cmd *icmd.Cmd) func() {
+	cmd.Command = append(cmd.Command, "--no-cache")
+	return nil
+}
+
+func withFile(name, content string) func(*FakeContext) error {
```

Comment:
All these functions are a bit weird in the main testing package. Maybe they need to be split out.